### PR TITLE
Travis python versions alternative

### LINKED ISSUE #93: Drop support of Python2?
Should we remove Python2.7/Pypy2.7 support, as it is reaching its end of life very soon (https://pythonclock.org/)?

Supporting 2.7 seems not obvious for some cases in Spark SQL (last Travis run: https://travis-ci.org/svenkreiss/pysparkling/builds/616898533?utm_source=github_status&utm_medium=notification),

Beside Apache Spark announced it won't support it in new versions: https://spark.apache.org/news/plan-for-dropping-python-2-support.html

One the other end we could add (C)Python 3.8 and Pypy3?

### PR DESCRIPTION
**Pull request by tools4origins**
This PR fixes #93 

It contains the same code modifications as #94 but has an alternative approach on git history side:

#94 keeps spark SQL git history, this PR does not, I believe it would be better to keep it as it contains valuable information but feel free to merge this one and close #94 so we can move forward :)

### CODE REVIEW

File: pysparkling/context.py [hunk 1]
```diff
@@ -23,7 +23,10 @@
 
 # Python 2 compatibility
```
Comment (svenkreiss) [inline pysparkling/context.py]: Should this whole block be removed now? Can also leave it for later cleanup.

File: pysparkling/tests/test_resolve_filenames.py [hunk 2]
```diff
@@ -68,6 +68,8 @@ def test_s3_1():
 
 
 def test_hdfs_resolve_filenames_with_wildcard():
+    # hdfs is an optional dependency
+    # pylint: disable=C0415
```
Comment (svenkreiss) [inline pysparkling/tests/test_resolve_filenames.py]: I'd prefer the human readable version here.

Comment (tools4origins) [reply pysparkling/tests/test_resolve_filenames.py]: Oh yes sure, I didn't know about it when I added the line but it's much better indeed, done!

File: pysparkling/context.py (hunk 1 above)
Comment (tools4origins) [reply pysparkling/context.py]: It is also possible to keep Python2 support in existing code, the effort to remove compatibility code does not seem worth it

**Pull request merged**